controllers/validation.js

```diff
@@ -0,0 +1,626 @@
+	const field = combinePropArgument(property, result[counter].argument);
+
+	output.errorArray.push(makeErrorObj(field, 'missing'));
+	findField(schema, field.split('.'), getAllRequired);
+	for (const i in requiredFields){
+		requiredFields[i] = `${field}.${requiredFields[i]}`;
+	}
+	requiredFields.forEach((requiredField)=>{
+		output.errorArray.push(makeErrorObj(requiredField, 'missing'));
+	});
+}
+
+function handleTypeError(output, result, counter){
+
+	const expectedType = result[counter].argument[0];
+	const property = removeInstance(result[counter].property);
+	output.errorArray.push(makeErrorObj(property, 'type', expectedType));
+
+}
+
+function handleFormatError(output, result, counter){
+
+	const field = `${removeInstance(result[counter].property)}`;
+	output.errorArray.push(makeErrorObj(field, 'format'));
+
+}
+
+function handleEnumError(output, result, counter){
+
+	const property = removeInstance(result[counter].property);
+	output.errorArray.push(makeErrorObj(property, 'enum', null, result[counter].message));
+
+}
+
+function getDependency(result, counter){
+
+	const stackMessage = result[counter].stack;
+	const dependency = stackMessage.split(' property ')[1].split(' not ')[0];
+	return dependency;
+
+}
+
+function handleDependencyError(output, result, counter){
+
+	const error = result[counter];
+	const dependentField = removeInstance(error.argument);
+	const schemaPath = removeInstance(error.property);
+	const dependency = `${schemaPath}.${getDependency(result, counter)}`;
+	output.errorArray.push(makeErrorObj(dependentField, 'dependencies', null, null, dependency));
+
+}
+
+/**
+ * Creates error object for errors resulting from an anyOf section of the validation schema
+ *
+ * @param {object} errorTracking - Error object containing all error to report and the error message to deliver.
+ * @param {array} errorTracking.errorArray - Array contain all errors to report to user.
+ * @param {array} result - Array of errors found during validation.
+ * @param {integer} counter - Position in result that the current error being handled is.
+ * 
+ * @affects errorTracking.errorArray 
+ */
+function handleAnyOfError(errorTracking, result, counter){
+
+	const error = result[counter];
+	const property = removeInstance(error.property);
+	const requiredOptions = [];
+	error.schema.anyOf.forEach((fieldObj)=>{
+		requiredOptions.push(combinePropArgument(property, fieldObj.required[0]));
+	});
+	errorTracking.errorArray.push(makeErrorObj(null, 'anyOf', null, null, null, requiredOptions));
+	
+}
+
+/** Get the schema to be used for validating user input
+ * @param  {Object} pathData - All data from swagger for the path that has been run
+ * @return {Object} schemas - fullSchema is the full validation schemas for all permit types. schemaToUse is the validation schema for this route
+ */
+function getValidationSchema(pathData){
+	const fileToGet = `controllers/${pathData.validation.$ref.split('#')[0]}`;
```

Comment:
Would like to figure out how to get the packages we are using to get this, rather than having to parse the field, pull in the schema ourselves and provide it to the package.  Pretty sure using schemas across multiple files is part of the standard, so it should be possible.